ViewModel/SwitcherUrlProvider.php

```diff
@@ -0,0 +1,196 @@
+            throw new \Exception('No story found');
+        }
+
+        $story = $story->getBody()['story'];
+        $hreflangs = array_key_exists('hreflangs', $story['content']) ? $story['content']['hreflangs'] : [];
+        $currentStore = $this->storeManager->getStore();
+        $stores = $this->storeManager->getStores();
+
+        $html = '';
+        foreach ($hreflangs as $relatedSlug) {
+            try {
+                $relatedStory = $this->storyblokClient->getStoryByUuid($relatedSlug);
+            } catch (\Exception $exception){
+                continue;
+            }
+
+            $this->generateSwitchItem($html, $relatedStory, $story, $currentStore, $stores);
+        }
+
+        if (empty($html)) {
+            throw new \Exception('No related stories found');
+        }
+
+        return $html;
+    }
+    private function getMagentoSwitchUrls(): string
+    {
+        $html = '';
+        $currentWebsiteId = $this->storeManager->getStore()->getWebsiteId();
+        foreach ($this->storeManager->getStores() as $store ) {
+            if (
+                !$store->isActive() ||
+                $store->getId() == $this->storeManager->getStore()->getId() ||
+                $currentWebsiteId !== $store->getWebsiteId()
+            ) {
+                continue;
+            }
+
+            $storeName = $store->getName();
+            $url = $this->encoder->encode($store->getCurrentUrl());
+            $url = $this->urlBuilder->getUrl(
+                'stores/store/redirect',
+                [
+                    '___store' => $store->getCode(),
+                    '___from_store' => $this->storeManager->getStore()->getCode(),
+                    ActionInterface::PARAM_NAME_URL_ENCODED => $url,
+                ]
+            );
+            $html .= $this->generateSwitchButton($url, $storeName);
+        }
+
+        return $html;
+    }
+
+    private function generateSwitchItem(string &$html, Client $relatedStory, array $story, $currentStore, $stores): void
+    {
+        if ($this->relatedStoryExists($relatedStory, $story)) {
+            $slug = $this->generateStoryblokSlug($relatedStory);
+            /** @phpstan-ignore-next-line */
+            $url = 'https://' . $this->request->getServer()->get('HTTP_HOST') . '/' . $slug . (str_ends_with($slug, '/') ? '' : '/');
+            foreach ($stores as $store) {
+                if (
+                    str_starts_with($url, $store->getBaseUrl()) &&
+                    $store->getWebsiteId() === $currentStore->getWebsiteId()
+                ) {
+                    $storeName = $store->getName();
+                    $html .= $this->generateSwitchButton($url, $storeName);
+                }
+            }
+        }
+    }
+
+    private function generateSwitchButton(string $url, string $storeName): string
+    {
+        return '<a href="' . $url . '" class="block px-4 py-2 rounded-md lg:px-5 lg:py-2 hover:bg-tertiary hover:underline underline-offset-2">
```

Comment:
Is this for a language switch?

If so, you could maybe use the `lang` attribute, eg:
```html
<ul>
  <li><a href="..." lang="de">Deutsch</a></li>
  <li><a href="..." lang="it">Italiano</a></li>
  <li><a href="..." lang="fr">Français</a></li>
  ```